gitlab/provider.go

```diff
@@ -108,20 +124,31 @@ func init() {
 		"client_cert": "File path to client certificate when GitLab instance is behind company proxy. File  must contain PEM encoded data.",
 
 		"client_key": "File path to client key when GitLab instance is behind company proxy. File must contain PEM encoded data.",
+
+		"fail_fast": "If false, enables the automatic deletion of resources within the Terraform state if the aforementioned managed resources where deleted in an out-of-band fashion, through the UI for example. NOTE that failing to provide a token with the appropriate permissions to access the resources WILL delete the resources from Terraform state, as Gitlab will try and preserve the resource anonymity from unauthorized users by returning a 'not found' response. In case of such a mistake, the resources will have to be re-imported. Use with caution. Defaults to true.",
```

Comment:
For me as a user, I would be less confused if an opt-in flag was something I had to set to `true`, not `false`. Maybe call it `detect_removed_resources`?

Another point - You write that "In case of such a mistake, the resources will have to be re-imported." But I thought that this flag is specifically for resources that _can't_ be imported.